data/block/miniBlockHeader_test.go

```diff
@@ -68,48 +89,133 @@ func TestMiniBlockHeader_SetConstructionState(t *testing.T) {
+	require.Equal(t, index-1, mbh.GetIndexOfFirstTxProcessed())
+	require.Equal(t, index, mbh.GetIndexOfLastTxProcessed())
 
-	state = int32(block.Final)
+	state = int32(block.PartialExecuted)
 	_ = mbh.SetConstructionState(state)
+	index = int32(78)
+	_ = mbh.SetIndexOfFirstTxProcessed(index - 1)
+	_ = mbh.SetIndexOfLastTxProcessed(index)
 
 	_ = mbh.SetProcessingType(int32(block.Scheduled))
 	require.Equal(t, state, mbh.GetConstructionState())
+	require.Equal(t, index-1, mbh.GetIndexOfFirstTxProcessed())
+	require.Equal(t, index, mbh.GetIndexOfLastTxProcessed())
 
-	_ = mbh.SetProcessingType(int32(block.Normal))
+	_ = mbh.SetProcessingType(int32(block.Processed))
 	require.Equal(t, state, mbh.GetConstructionState())
+	require.Equal(t, index-1, mbh.GetIndexOfFirstTxProcessed())
+	require.Equal(t, index, mbh.GetIndexOfLastTxProcessed())
 }
 
-func TestMiniBlockHeader_SetConstructionStateDoesNotChangeProcessingType(t *testing.T) {
+func TestMiniBlockHeader_SetConstructionStateDoesNotChangeProcessingTypeOrIndexOfTxProcessed(t *testing.T) {
 	mbh := &block.MiniBlockHeader{}
 	processingType := int32(block.Scheduled)
 	_ = mbh.SetProcessingType(processingType)
+	index := int32(21)
+	_ = mbh.SetIndexOfFirstTxProcessed(index - 1)
+	_ = mbh.SetIndexOfLastTxProcessed(index)
 
 	_ = mbh.SetConstructionState(int32(block.Proposed))
-	require.Equal(t, processingType, mbh.GetConstructionState())
+	require.Equal(t, processingType, mbh.GetProcessingType())
+	require.Equal(t, index-1, mbh.GetIndexOfFirstTxProcessed())
+	require.Equal(t, index, mbh.GetIndexOfLastTxProcessed())
 
-	_ = mbh.SetConstructionState(int32(block.Final))
+	_ = mbh.SetConstructionState(int32(block.PartialExecuted))
 	require.Equal(t, processingType, mbh.GetProcessingType())
+	require.Equal(t, index-1, mbh.GetIndexOfFirstTxProcessed())
+	require.Equal(t, index, mbh.GetIndexOfLastTxProcessed())
 
-	processingType = int32(block.Normal)
+	processingType = int32(block.Processed)
 	_ = mbh.SetProcessingType(processingType)
+	index = int32(78)
+	_ = mbh.SetIndexOfFirstTxProcessed(index - 1)
+	_ = mbh.SetIndexOfLastTxProcessed(index)
 
 	_ = mbh.SetConstructionState(int32(block.Proposed))
 	require.Equal(t, processingType, mbh.GetProcessingType())
+	require.Equal(t, index-1, mbh.GetIndexOfFirstTxProcessed())
+	require.Equal(t, index, mbh.GetIndexOfLastTxProcessed())
 
-	_ = mbh.SetConstructionState(int32(block.Final))
+	_ = mbh.SetConstructionState(int32(block.PartialExecuted))
+	require.Equal(t, processingType, mbh.GetProcessingType())
+	require.Equal(t, index-1, mbh.GetIndexOfFirstTxProcessed())
+	require.Equal(t, index, mbh.GetIndexOfLastTxProcessed())
+}
+
+func TestMiniBlockHeader_SetIndexOfTxProcessedDoesNotChangeConstructionStateOrProcessingType(t *testing.T) {
+	mbh := &block.MiniBlockHeader{}
+	processingType := int32(block.Scheduled)
+	_ = mbh.SetProcessingType(processingType)
+	state := int32(block.Proposed)
+	_ = mbh.SetConstructionState(state)
+
+	_ = mbh.SetIndexOfFirstTxProcessed(20)
```

Comment:
please also add tests for the bounders (index 0, index -1, index max int32, and so on)

Reply:
Done